timeseries/tests/unittests/test_features.py

```diff
@@ -169,3 +174,252 @@ def test_when_static_features_contain_missing_values_then_they_are_filled_during
 
     data_transformed = feat_generator.fit_transform(data)
     assert not data_transformed.static_features.isna().any(axis=None)
+
+
+@pytest.mark.parametrize(
+    "item_id_to_length",
+    [
+        {"A": 10, "B": 15},
+        {"A": 10, "B": 10, "C": 10},
```

Comment:
To be extra safe, use item_ids that are not sorted alphabetically (e.g. B, C, A) to avoid weird bugs where groupby accidentally sorts the index.